src/js/tracks/audio-track-list.js

```diff
@@ -72,22 +72,34 @@ class AudioTrackList extends TrackList {
       return;
     }
 
+    if (!this.enabledChange_) {
+      this.enabledChange_ = () => {
```

Comment:
looks like this change causes the first track to always get selected because we aren't changing it when new tracks come in. https://github.com/videojs/video.js/issues/5887

Probably similar issue in video tracks and text tracks.